### PR TITLE
1041 ensure handle link is absolute url

### PR DESCRIPTION
**Pull request by hannahtuttle**
This adds validation to the link input in the warden-registration, it also changed the placeholder in the input field and updates a few warden links to be absolute links. Needs to wait for warden auth before changes can be made.

### CODE REVIEW

File: src/components/RegistrationForm.tsx [hunk 1]
```diff
@@ -102,6 +103,18 @@ export default function RegistrationForm({ handles, wardens, className }) {
...
+    const check = new RegExp('^(?:[a-z]+:)?//', 'i');
+    if(link !== undefined){
+      if(!check.test(link) && link !== ''){
+        setHasValidateLinkError(true);
+        return;
+      }
+    }
+    setHasValidateLinkError(false);
+  },[state.link]);
+
```
Comment (Simon-Busch) [inline src/components/RegistrationForm.tsx:117]:
```suggestion
```

I would simply handle is the same way we handle the discord username validation, see here:

```javascript
  const handleChange = useCallback((e) => {
    const { name, value } = e.target;
    setState((prevState) => {
      return { ...prevState, [name]: value };
    });
    if (name === "discordUsername") {
      setIsValidDiscord(discordUsernameRegex.test(value));
    }
    if (name === "username") {
      setisDangerousUsername(value.match(/^[0-9a-zA-Z_\-]+$/) === null);
    }
  }, []);
```

**Review by Simon-Busch — CHANGES_REQUESTED** [review]

**Review by Simon-Busch — APPROVED** [review]

File: src/components/RegistrationForm.tsx [hunk 2]
```diff
@@ -64,11 +70,13 @@ export default function RegistrationForm({ handles, wardens, className }) {
   const [status, setStatus] = useState<FormStatus>(FormStatus.Unsubmitted);
   const [errorMessage, setErrorMessage] = useState<string | ReactNode>("");
   const [isDangerousUsername, setisDangerousUsername] = useState(false);
+  const [hasValidateLinkError, setHasValidateLinkError] = useState(false);
   const [captchaToken, setCaptchaToken] = useState("");
 
   // global variables
   const avatarInputRef = useRef<HTMLInputElement>();
   const discordUsernameRegex = new RegExp(/.*#[0-9]{4}/, "g");
+  const checkAbsoluteURL = new RegExp("^(?:[a-z]+:)?//", "i");
```
Comment (joshuashort) [inline src/components/RegistrationForm.tsx:79]: Specifically http(s)?

```suggestion
  const checkAbsoluteURL = new RegExp("^(http[s]?)://", "i");
```

File: src/components/RegistrationForm.tsx [hunk 3]
```diff
@@ -503,6 +518,11 @@ export default function RegistrationForm({ handles, wardens, className }) {
                   value={state.link}
                   onChange={handleChange}
                 />
+                {hasValidationErrors && hasValidateLinkError && (
+                  <p className={widgetStyles.ErrorMessage}>
+                    Please provide a valid url. "https://domain_name" required.
```
Comment (joshuashort) [inline src/components/RegistrationForm.tsx:523]:
```suggestion
                    Please provide a valid http(s) url.
```

**Comment by Simon-Busch** [comment]
As discussed with @hannahtuttle We can close this P.R.

**Pull request closed without merging**